client/client.go

```diff
@@ -99,6 +99,16 @@ type OpenstackClient struct {
 	controllerID string
 }
 
+func NewTestOpenStackClient(mockClient *gophercloud.ServiceClient, controllerID string) *OpenstackClient {
```

Comment:
This should probably be moved to the _test.go file.

Reply:
From what I have tried before, the functions under `file_test.go` will not be seen by go as part of that package. So, if I move that function to client_test.go, it wont be seen by the provider.go (where it is used) as part of the client package

Reply:
you can use something like this:

https://github.com/cloudbase/garm/blob/6e416bb4947380735771bd1234b90fe63c6e46d4/internal/testing/testing.go#L15-L16

and pass that tag when running the tests:

https://github.com/cloudbase/garm/blob/6e416bb4947380735771bd1234b90fe63c6e46d4/Makefile#L91

Reply:
Created a new file testing.go in the client package that is accessed only by the flag testing.